var/spack/repos/builtin/packages/papi/package.py

```diff
@@ -42,53 +42,58 @@ class Papi(Package):
 
     depends_on('lm-sensors', when='+lmsensors')
 
+    conflicts('%gcc@8:', when='@5.3.0', msg='Requires GCC version less than 8.0')
+
     # Does not build with newer versions of gcc, see
     # https://bitbucket.org/icl/papi/issues/46/cannot-compile-on-arch-linux
     patch('https://bitbucket.org/icl/papi/commits/53de184a162b8a7edff48fed01a15980664e15b1/raw', sha256='64c57b3ad4026255238cc495df6abfacc41de391a0af497c27d0ac819444a1f8', when='@5.4.0:5.6.99%gcc@8:')
 
-    def setup_build_environment(self, env):
-        if '+lmsensors' in self.spec and self.version >= Version('6'):
-            env.set('PAPI_LMSENSORS_ROOT', self.spec['lm-sensors'].prefix)
+    configure_directory = 'src'
 
-    def setup_run_environment(self, env):
+    def setup_lmsensors(self, env):
         if '+lmsensors' in self.spec and self.version >= Version('6'):
             env.set('PAPI_LMSENSORS_ROOT', self.spec['lm-sensors'].prefix)
 
-    def install(self, spec, prefix):
-        if '+lmsensors' in spec:
-            if self.version < Version('6'):
-                with working_dir("src/components/lmsensors"):
-                    configure_args = [
-                        "--with-sensors_incdir=%s/sensors" %
-                        spec['lm-sensors'].headers.directories[0],
-                        "--with-sensors_libdir=%s" %
-                        spec['lm-sensors'].libs.directories[0]]
-                    configure(*configure_args)
-        with working_dir("src"):
-
-            configure_args = ["--prefix=%s" % prefix]
-
-            # PAPI uses MPI if MPI is present; since we don't require
-            # an MPI package, we ensure that all attempts to use MPI
-            # fail, so that PAPI does not get confused
-            configure_args.append('MPICC=:')
-
-            configure_args.append(
-                '--with-components={0}'.format(' '.join(
-                    filter(lambda x: spec.variants[x].value, spec.variants))))
-
-            configure(*configure_args)
-
-            # Don't use <malloc.h>
-            for level in [".", "*", "*/*"]:
-                files = glob.iglob(join_path(level, "*.[ch]"))
-                filter_file(r"\<malloc\.h\>", "<stdlib.h>", *files)
-
-            make()
-            make("install")
-
-            # The shared library is not installed correctly on Darwin
-            if sys.platform == 'darwin':
-                os.rename(join_path(prefix.lib, 'libpapi.so'),
-                          join_path(prefix.lib, 'libpapi.dylib'))
-                fix_darwin_install_name(prefix.lib)
+    def setup_build_environment(self, env):
+        self.setup_lmsensors(env)
+
+    def setup_run_environment(self, env):
+        self.setup_lmsensors(env)
```

Comment:
You could instead implement things in `setup_build_environment` and then add:
```python
setup_run_environment = setup_build_environment
```

Reply:
Thanks for the suggestion Adam.